app/assets/javascripts/TT.js

```diff
@@ -164,22 +164,33 @@ var TT = function() {
 
     var duration = Math.round(mins*60);
     var startDate = new Date();
+    var pausedDate = new Date();
     var timer = duration;
     stateStart(timer);
 
     (function tick() {
-      var msPassed = new Date() - startDate;
-      timer = Math.round(duration - msPassed/1000);
-
-      stateCounting(timer, duration);
+      var currentStatus = getStatus();
+      if('pause_on' == currentStatus) {
+        pausedDate = new Date();
+        status = 'paused';
+      }
+      else if('pause_off' == currentStatus) {
+        startDate = new Date() - (pausedDate - startDate) + 1000;
+        status = 'running';
+      }
+      if('running' == currentStatus) { 
```

Comment:
I think here you should check `getStatus()` or `status` otherwise you'd be losing a tick. If I'm correct the same would apply on line 193.